packages/core/src/app/customer/CheckoutButtonListV2.tsx

```diff
@@ -0,0 +1,54 @@
+import React, { Fragment, FunctionComponent } from 'react';
+
+import { withCheckout, WithCheckoutProps } from '../checkout';
+import { TranslatedString, withLanguage, WithLanguageProps } from '../locale';
+import resolveCheckoutButton from './resolveCheckoutButton';
+import CheckoutButton from './CheckoutButtonV2';
+
+export interface CheckoutButtonListProps {
+    onUnhandledError(error: Error): void;
+}
+
+const CheckoutButtonList: FunctionComponent<
+    CheckoutButtonListProps &
+    WithCheckoutProps &
+    WithLanguageProps
+> = ({
+    checkoutService,
+    checkoutState,
+    language,
+    onUnhandledError
+}) => {
+    const { 
+        statuses: { isInitializingCustomer },
+        data: { getConfig },
+    } = checkoutState;
```

Comment:
Just out of curiosity, we usually do the pattern from other examples that we do the `mapToXYZComponentProps`. Is that something we will avoiding going ahead? 

Does it make sense going ahead to move this setup into a custom hook/context and use that?

Reply:
We spoke offline. I'm fine with switching to using custom hooks rather than HOC. But currently we don't have an equivalent hook to use.

Regarding the props mapper, if there's nothing to filter, there's no need to write one. In this case, it's a simple passthrough because we need to propagate both the `checkoutService` and `checkoutState` to the inner component.